### PR TITLE
Support for bracket styled posargs

### LINKED ISSUE #1928: tox 4 bracket posargs are missing
as example i cut down the apipkg tox.ini to just

```
[testenv]
deps=pytest
commands=pytest []
```
ran
```
$ tox4
py: install_deps> python -I -m pip install pytest
.pkg: get_requires_for_build_wheel> python /home/rpfannsc/.local/lib/python3.9/site-packages/tox/util/pep517/backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /home/rpfannsc/.local/lib/python3.9/site-packages/tox/util/pep517/backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /home/rpfannsc/.local/lib/python3.9/site-packages/tox/util/pep517/backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /home/rpfannsc/.local/lib/python3.9/site-packages/tox/util/pep517/backend.py True setuptools.build_meta __legacy__
py: install_package> python -I -m pip install --no-deps --force-reinstall /home/rpfannsc/Projects/pytest-dev/apipkg/.tox/4/.pkg/dist/apipkg-1.6.dev16+ge54efbf.d20210227.tar.gz
py: commands[0]> pytest '[]'
ERROR: file or directory not found: []

========================================================================================= test session starts ==========================================================================================
platform linux -- Python 3.9.1, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
apipkg full install version=1.6.dev16+ge54efbf.d20210227
rootdir: /home/rpfannsc/Projects/pytest-dev/apipkg
collected 0 items

======================================================================================== no tests ran in 0.00s =========================================================================================
py: exit 4 (0.34 seconds) /home/rpfannsc/Projects/pytest-dev/apipkg> pytest '[]' pid=1183180
.pkg: _exit> python /home/rpfannsc/.local/lib/python3.9/site-packages/tox/util/pep517/backend.py True setuptools.build_meta __legacy__
  py: FAIL code 4 (9.90=setup[9.57]+cmd[0.34] seconds)
  evaluation failed :( (10.00 seconds)
```
and ran 
```
$ tox
GLOB sdist-make: /home/rpfannsc/Projects/pytest-dev/apipkg/setup.py
python create: /home/rpfannsc/Projects/pytest-dev/apipkg/.tox/python
python installdeps: pytest
python inst: /home/rpfannsc/Projects/pytest-dev/apipkg/.tox/.tmp/package/1/apipkg-1.6.dev16+ge54efbf.d20210227.zip
python installed: apipkg @ file:///home/rpfannsc/Projects/pytest-dev/apipkg/.tox/.tmp/package/1/apipkg-1.6.dev16%2Bge54efbf.d20210227.zip,attrs==20.3.0,iniconfig==1.1.1,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyparsing==2.4.7,pytest==6.2.2,toml==0.10.2
python run-test-pre: PYTHONHASHSEED='1155200767'
python run-test: commands[0] | pytest
========================================================================================= test session starts ==========================================================================================
platform linux -- Python 3.9.1, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
cachedir: .tox/python/.pytest_cache
apipkg full install version=1.6.dev16+ge54efbf.d20210227
rootdir: /home/rpfannsc/Projects/pytest-dev/apipkg
collected 42 items                                                                                                                                                                                     

test_apipkg.py ..........................................                                                                                                                                        [100%]

========================================================================================== 42 passed in 0.63s ==========================================================================================
_______________________________________________________________________________________________ summary ________________________________________________________________________________________________
  python: commands succeeded
  congratulations :)
```
and then ran
```
$ tox -- -k aliasmod
GLOB sdist-make: /home/rpfannsc/Projects/pytest-dev/apipkg/setup.py
python inst-nodeps: /home/rpfannsc/Projects/pytest-dev/apipkg/.tox/.tmp/package/1/apipkg-1.6.dev16+ge54efbf.d20210227.zip
python installed: apipkg @ file:///home/rpfannsc/Projects/pytest-dev/apipkg/.tox/.tmp/package/1/apipkg-1.6.dev16%2Bge54efbf.d20210227.zip,attrs==20.3.0,iniconfig==1.1.1,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyparsing==2.4.7,pytest==6.2.2,toml==0.10.2
python run-test-pre: PYTHONHASHSEED='1148858278'
python run-test: commands[0] | pytest -k aliasmod
========================================================================================= test session starts ==========================================================================================
platform linux -- Python 3.9.1, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
cachedir: .tox/python/.pytest_cache
apipkg full install version=1.6.dev16+ge54efbf.d20210227
rootdir: /home/rpfannsc/Projects/pytest-dev/apipkg
collected 42 items / 33 deselected / 9 selected                                                                                                                                                        

test_apipkg.py .........                                                                                                                                                                         [100%]

=================================================================================== 9 passed, 33 deselected in 0.19s ===================================================================================
_______________________________________________________________________________________________ summary ________________________________________________________________________________________________
  python: commands succeeded
  congratulations :)

```

### PR DESCRIPTION
**Pull request by hexagonrecursion**
Fixes #1928
In #1950 me and @gaborbernat have discussed how the `[]` substitution should behave in tox4. In this PR I'll implement the changes.

- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [X] added yourself to `CONTRIBUTORS` (preserving alphabetical order)
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
- [x] added the tests
- [x] implemented the syntax

I assume adding myself to `CONTRIBUTORS` on the master branch counts.

### CODE REVIEW

File: docs/changelog/1928.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Implemented [] substitution (alias for {posargs})
```
Comment (gaborbernat) [inline docs/changelog/1928.feature.rst:1]: quote both ``[]`` and ``{posargs}`` also add your user name as others do.

File: tests/config/loader/ini/replace/test_replace_posargs.py [hunk 2]
```diff
@@ -5,18 +5,27 @@
 from tests.config.loader.ini.replace.conftest import ReplaceOne
 
 
-def test_replace_pos_args_none_sys_argv(replace_one: ReplaceOne) -> None:
-    result = replace_one("{posargs}", None)
+@pytest.mark.parametrize(
+    "syntax", ["{posargs}", pytest.param("[]", marks=pytest.mark.xfail(raises=AssertionError))]  # noqa: SC200
```
Comment (gaborbernat) [inline tests/config/loader/ini/replace/test_replace_posargs.py:9]: don't silence error instead add to whitelist https://github.com/tox-dev/tox/blob/rewrite/whitelist.txt

Comment (gaborbernat) [reply tests/config/loader/ini/replace/test_replace_posargs.py:9]: I'm not a fan of all this noise on the PRs. Can you just make a final commit when you're close to done?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/config/loader/ini/replace/test_replace_posargs.py (hunk 2 above)
Comment (gaborbernat) [reply tests/config/loader/ini/replace/test_replace_posargs.py:9]: Development should happen locally, the CI should only be used to do the final validation (let's not overload Github system needlessly).

File: docs/changelog/1928.feature.rst (hunk 1 above)
Comment (hexagonrecursion) [reply docs/changelog/1928.feature.rst]: done

**Comment by gaborbernat** [comment]
@hexagonrecursion in the future can you please give human-readable titles to the PRs, please 🙏🏻

**Review by gaborbernat — APPROVED** [review]
Made some changes that I think simplifies the implementation a bit and gives the same outcome (at least according to the current test suite).

**Pull request merged**